src/js/components/HandleSearchError.js

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Link } from 'react-router-dom';
+import { ACTIONS, errorEvent } from '../util/analytics';
+import GenericError, { SachKaur } from './GenericError';
+import { TYPES, SOURCES, PLACEHOLDERS, TEXTS } from '../constants';
+
+export default class HandleSearchError extends React.PureComponent {
+  static propTypes = {
+    q: PropTypes.string.isRequired,
+    type: PropTypes.number.isRequired,
+    source: PropTypes.string.isRequired,
+    children: PropTypes.node.isRequired,
```

Comment:
Wasn't sure about the type of "children" but read this: https://stackoverflow.com/questions/42122522/reactjs-what-should-the-proptypes-be-for-this-props-children and settled on "node".